### PR TITLE
MSVCx86: use pretty names instead of mangled names for exceptions

### PR DESCRIPTION
**Pull request by rainers**
This is necessary because the mangled name cannot always be created from the type info at runtime (e.g. exceptions defined as function local symbols contain the function signature in the mangled name, but not in the TypeInfo).
Two other advantages:
- code is simpler
- it looks better in the debugger...

### CODE REVIEW

File: src/ldc/eh/win32.d [hunk 1]
```diff
@@ -79,9 +79,9 @@ extern(Windows) void RaiseException(DWORD dwExceptionCode,
                                     DWORD nNumberOfArguments,
                                     ULONG_PTR* lpArguments);
 
-__gshared TypeDescriptor!16 tdObject    = { 0, null, "_D6object6Object\0" };
-__gshared TypeDescriptor!20 tdThrowable = { 0, null, "_D6object9Throwable\0" };
-__gshared TypeDescriptor!12 tdException = { 0, null, "_D9Exception\0" };
+__gshared TypeDescriptor!16 tdObject    = { 0, null, "Dobject.Object\0" };
+__gshared TypeDescriptor!19 tdThrowable = { 0, null, "Dobject.Throwable\0" };
+__gshared TypeDescriptor!19 tdException = { 0, null, "Dobject.Exception\0" };
```
Comment (kinke) [inline src/ldc/eh/win32.d]: Shouldn't these be 14, 17 and 17?

File: src/ldc/eh/win32.d [hunk 2]
```diff
@@ -155,56 +155,16 @@ CatchableType* getCatchableType(TypeInfo_Class ti)
...
-            memcpy(buf, s.ptr + q, p - q);
-            buf += p - q;
-            q = p + 1;
-        }
-        *buf = 0;
-    }
-
-    size_t mangledLength = mangledNameLength(ti.name) + 1;
-    size_t sz = TypeDescriptor!1.sizeof + mangledLength;
+    size_t sz = TypeDescriptor!1.sizeof + ti.name.length + 2;
```
Comment (kinke) [inline src/ldc/eh/win32.d]: `+ 2` is obsolete here I think - `TypeDescriptor!1.name` already provides space for 2 chars (`D` prefix and `\0` suffix).

File: src/ldc/eh/win32.d (hunk 1 above)
Comment (rainers) [inline src/ldc/eh/win32.d]: Yes. I remember having counted the length of "object" as 7 ;-)

File: src/ldc/eh/win32.d (hunk 2 above)
Comment (rainers) [inline src/ldc/eh/win32.d]: Agree. It was too large in the previous version, too.

**Comment by rainers** [comment]
I've removed the predefined TypeDescriptor records, the full ThrowInfo include base types is now preallocated for Exception and OutOfMemoryError.

**Pull request merged**